### PR TITLE
Areas can have their own base turf type

### LINKED ISSUE #10399: Tomb away mission breaches into space when an explosive is detonated.
#### Description of issue

Floor can be breached.
#### Difference between expected and actual behavior

A desert shouldn't breach into space when an explosion occurs.
#### Steps to reproduce

Detonate a bomb on the ground.
#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by unid15**
- Areas now have a "base_turf_type" variable
- It overrides the z-level's base turf type
- Added a new "get_underlying_turf" proc for turfs
- Tomb of Rafid no longer creates space when blown up

Fixes #10399

### CODE REVIEW

**Comment by Konater** [comment]
fuck yes this will be great for shuttles

**Comment by ghost** [comment]
Shuttles already have a base turf though too

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -503,8 +503,15 @@
 				S.air.update_values()
 */
 
+/turf/proc/get_underlying_turf()
+	var/area/A = loc
+	if(A.base_turf_type)
+		return A.base_turf_type
+
+	return get_base_turf(z)
```
Comment (ghost) [inline code/game/turfs/turf.dm:511]: Does this break shuttle docking base turfs?

**Comment by Konater** [comment]
I mean uh, the turf they leave behind when they move
The shuttles at central command/nukeop ship are good examples

**Comment by unid15** [comment]
@Konater that already exists, and it's not done by areas, but rather by docking ports

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (unid15) [inline code/game/turfs/turf.dm:511]: I don't see how it would

**Comment by Konater** [comment]
huh

**Comment by clusterfack** [comment]
Yay I love it when my ideas are done by other people

**Comment by The-Albinobigfoot** [comment]
> I was procrastinating on porting polaris' station due to this.

:ok_hand:

**Pull request merged**